redirect/redirect.c

```diff
@@ -0,0 +1,59 @@
+/*
+* SPDX-FileCopyrightText: (c) 2024 Ring Zero Desenvolvimento de Software LTDA
+* SPDX-License-Identifier: GPL-2.0-only
+*/
+
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+
+#define ETH_P_IP 0x0800
+#define ETH_ALEN 6
+
+#define memcmp __builtin_memcmp
+#define memcpy __builtin_memcpy
+
+static const u32 eth1 = 3;
+static const u32 eth2 = 4;
+
+static const u8 macaddr_gw1[ETH_ALEN]  = {0x08, 0x00, 0x27, 0xb0, 0x90, 0x8f};
+static const u8 macaddr_gw2[ETH_ALEN]  = {0x08, 0x00, 0x27, 0xca, 0x8c, 0x88};
+static const u8 macaddr_cli[ETH_ALEN]  = {0x08, 0x00, 0x27, 0x65, 0xdd, 0x94};
+static const u8 macaddr_dome[ETH_ALEN] = {0x08, 0x00, 0x27, 0xa2, 0xee, 0xcc};
```

Comment:
we shouldn't hard code the mac addresses, right? the only way to parametrize this is through maps? if so, I agree we can do this in a new PR..

Reply:
not necessarily the only way I can think of, but surely the best. I intend to jump right into it after I'm done with the benchmarks, and I don't expect it to take long, so, it's up to you.

Reply:
I totally agreed to do this later.. however, I think we could at least move this to something like `config.h`